### ISSUE TITLE
Data always saved as int16

### ISSUE THREAD

**Issue by rharkes** [nvladimus/npy2bdv#10]
It would be nice if npy2bdv would also offer the option to store data in a different datatype than int16. Maybe the writer could give a warning if data might have been altered? This is the case for any data not in `[np.uint8 np.int8 np.int16]`.
The code below shows that if you have data in float32, uin16 or uint32 the values in the resulting file are different from the values that have gone in. 
```
import h5py
import npy2bdv
import numpy as np
from pathlib import Path
pth = Path('c:\\temp\\testfile.h5')
# bitdepth = [8, np.float32]
# bitdepth = [16, np.uint16]
bitdepth = [32, np.uint32]
if isinstance(bitdepth[1](), np.floating):
    dset = np.random.rand(100, 100, 100).astype(bitdepth[1])
else:
    dset = np.random.randint(2 ** bitdepth[0], size=(100, 100, 100), dtype=bitdepth[1])
bdv_writer = npy2bdv.BdvWriter(str(pth), overwrite=True)
bdv_writer.append_view(dset)
bdv_writer.write_xml()
bdv_writer.close()
f = h5py.File(pth, 'r')
print(f"data type = {f['t00000']['s00']['0']['cells'].dtype}")
print(f"maximum in dataset = {np.max(dset)}")
print(f"maximum in file = {np.max(f['t00000']['s00']['0']['cells'])}")
f.close()
```

**Comment by nvladimus** [comment]
Hi, @rharkes 
The question is whether other formats are used in BigDataViewer for microscopy images. From my experience, microscopy data is almost always uint16. What's your application field? What format do you actually need?

**Comment by rharkes** [comment]
I ran into the problem when binning multiple pixels together. The value of the sum went over 65535 so I thought to store them as uint32. But I agree this is probably an exception. However, after some analysis has been applied, data is often in float32 format. For example after convolution filters. Also we very often have multi channel data from confocal microscopes with 8 bit pixels per channel. There it is a bit of a waste to store and open them as uint16 I would say.

**Comment by nvladimus** [comment]
I agree that float32 makes sense to add. I am unfortunately quite busy at the moment to add and test this properly. As for `int8` that some confocal scopes generate by default, I consider this a micro-crime, and it should be reported to microscopy police 👮 🔬
